tests/hazmat/primitives/test_dh.py

```diff
@@ -18,6 +19,11 @@
 from ...utils import load_nist_vectors, load_vectors_from_file
 
 
+skip_no_dhx = pytest.mark.skipif(
```

Comment:
So really if q is present and the user attempts to serialize or deserialize we should raise some sort of unsupported exception?